src/extensions/agenda.py

```diff
@@ -100,52 +99,30 @@ async def gen_agenda(
     formatted_time = parsed_datetime.strftime("%H:%M")
     formatted_datetime = parsed_datetime.strftime("%A, %Y-%m-%d %H:%M")
 
-    if "https://md.redbrick.dcu.ie" not in url:
+    try:
+        content = await get_md_content(url, aiohttp_client)
+    except Exception as e:
```

Comment:
Use `aiohttp.ClientResponseError` here for an error with the request, and add another except block for `ValueError` when the URL is invalid.